vllm/model_executor/layers/fused_moe/fused_moe.py

```diff
@@ -420,13 +421,10 @@ def fused_experts(hidden_states: torch.Tensor,
         torch.float32, torch.float16, torch.bfloat16
     ]
 
-    M, _ = hidden_states.shape
+    num_tokens, _ = hidden_states.shape
     E, N, _ = w1.shape
-
-    if M > 65536:
-        # https://github.com/vllm-project/vllm/issues/5938
-        raise ValueError("MoE kernel does not support more than 65536 tokens, "
-                         f"but got {M}")
+    CHUNK_SIZE = envs.VLLM_FUSED_MOE_CHUNK_SIZE
```

Comment:
Add a comment linking to the issue for the reason we do this chunk computation.

Reply:
Regarding the unit test, wdyt about simply adding cases to the parameterized test_fused_moe() in test_moe.py where longer inputs will be tested?

Reply:
Yeah that's what I have in mind. Just make sure we can handle longer inputs fixed by this PR.